python/tk_framework_desktopserver/shotgun/api_v2.py

```diff
@@ -0,0 +1,758 @@
+# Copyright (c) 2017 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+import sys
+import os
+import cPickle
+import sqlite3
+import json
+import subprocess
+import tempfile
+import contextlib
+import json
+import fnmatch
+import datetime
+
+import sgtk
+from sgtk.commands.clone_configuration import clone_pipeline_configuration_html
```

Comment:
You can basicaly ask the tk instance for the `clone_configuration` command. it's interface might be as nice as what you have for the HTML stuff however, hence why I said it might need to finessing.